locals.tf

```diff
@@ -5,6 +5,6 @@ locals {
   cognito_user_pool   = toset(var.apigateway.cognito_user_pool_name != null ? [var.apigateway.cognito_user_pool_name] : [])
   function_name       = var.resource_prefix != null ? "${var.resource_prefix}__${var.lambda.name}" : var.lambda.name
 
-  subnet_ids         = var.lambda.network.subnets_tag != null ? data.aws_subnets.functions.ids : var.vpc_cidr_block != null ? [aws_subnet.public_subnet[0].id, aws_subnet.private_subnet[0].id] : null
-  security_group_ids = var.lambda.network.security_groups_tag != null ? data.aws_security_groups.functions.ids : var.security_group_name != null ? [aws_security_group.lambda[0].id] : null
+  subnet_ids         = var.lambda.network.subnets_tag != null ? data.aws_subnets.functions[0].ids : var.vpc_cidr_block != null ? [aws_subnet.public_subnet[0].id, aws_subnet.private_subnet[0].id] : null
+  security_group_ids = var.lambda.network.security_groups_tag != null ? data.aws_security_groups.functions[0].ids : var.security_group_name != null ? [aws_security_group.lambda[0].id] : null
```

Comment:
Fala Gab! Mais por curiosidade, consegue explicar o que esse trecho de código faz?

Reply:
Fala Gui, sem problemas!

Ambas as variáveis estão sendo construídas da seguinte forma:

1. Verificamos se existem tags para se utilizar de uma VPC já existente `var.lambda.network.subnets_tag != null` e `var.lambda.network.security_groups_tag`
2. Caso essas tags tenham sido definidas, então buscamos na AWS através do `data`os recursos correspondentes a essas tags `data.aws_subnets.functions[0].ids` e `data.aws_security_groups.functions[0].ids`
3. Agora caso essas tags não existam, então verificamos as variáveis para criação `var.vpc_cidr_block != null` e `var.security_group_name != null`
4. No caso dessas variáveis terem sido definidas, então criamos e recuperamos as informações dos recursos `[aws_subnet.public_subnet[0].id, aws_subnet.private_subnet[0].id]` e `[aws_security_group.lambda[0].id]`
5. E no cenário onde nem as tags e nem as variáveis para criação de uma nova VPC tenham sido definidas, então simplemente deixamos como `null`

Para fazer um resumão:

Verificamos se temos como recuperar uma VPC já existente através de tags -> Caso existam, utilizamos os dados da VPC encontrada -> Caso contrário verificamos a possibilidade de se criar uma nova VPC -> Com a VPC criada, recuperamos seus dados -> E no fim, se não for possível recuperar ou criar esses recurso, deixamos o valor como `null`